ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFFromUnixTime.java

```diff
@@ -107,6 +107,10 @@ public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumen
     }
 
     if (timeZone == null) {
+      if (SessionState.get() == null) {
+        SessionState ss = new SessionState(new HiveConf());
+        SessionState.setCurrentSessionState(ss);
```

Comment:
please don't start a new session here
* in case you don't have a session - you may read this info from a `new HiveConf()`
* bumping into this issue from the HMS side is kinda unexpected - could you please leave a note about why this is needed here
* could you cover this issue with a testcase - or that's not possible because we are running embedded metastore in the tests?

Reply:
* sorry I did not follow this correctly, I think SessionState will not be available in HMS, can you please guide me how can I retrieve it from new HiveConf()
* yes that's a rare case while filtering the partition by expression, a query with filter condition having timestamp/date as predicate e.g. following query
select * from table_name where hour between from_unixtime(unix_timestamp('2020090120', 'yyyyMMddHH') - 1*60*60, 'yyyyMMddHH') and from_unixtime(unix_timestamp('2020090122', 'yyyyMMddHH') + 2*60*60, 'yyyyMMddHH');

in case of  PartitionExpressionForMetastore proxy, the filter expression is based on ql class/package and thats where user will see such exception.

* I think its difficult to reproduce with embedded HMS.